forem/forem
# Fix legacy sitemap logic

## What type of PR is this? (check all applicable)


## Description

This builds on https://github.com/forem/forem/pull/14857 to add a proper sitemap index to `robots.txt` and remove the way of doing things via `sitemap_generator` which never fully supported all the contexts for Forem and abstracts away some logic which is more straightforward if we do without it.

Basically, the sitemap we were serving in `robots.txt` was not really the way Google wants it and fully wasn't working in Forem Cloud or self-hosted Forems.

This changes should be better for DEV as well as make this work properly for other Forems.

This basically maintains the logic the current sitemap, and I think it is more or less appropriate. It orders posts by publish date, but tags and users more by total activity, as they are more of landing pages and "newness" isn't as much of a thing. We could revisit some of this. The underlying functionality is not yet documented in admin docs, so I could see giving this another pass before setting our documented logic in stone.

## Related Tickets & Documents

https://github.com/forem/forem/pull/14857

## QA Instructions, Screenshots, Recordings

<img width="554" alt="Screen Shot 2021-10-04 at 3 46 19 PM" src="https://user-images.githubusercontent.com/3102842/135914635-a1400748-31f5-4da5-98b9-50cc8d6aa357.png">

<img width="512" alt="Screen Shot 2021-10-04 at 3 46 36 PM" src="https://user-images.githubusercontent.com/3102842/135914683-6f9e3316-3bf9-4ada-b9f9-0e9350359a7f.png">

Examine the logic and make sure everything basically makes sense.

Check out this page to double-check that everything seems compatible:

https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

Not yet — maybe we document this with some best practices in near future.
